### PR TITLE
chore(config): migrate ClusterAgentConfiguration to typed config

### PR DESCRIPTION
**Pull request by webern**
## AI Summary

Migrates `ClusterAgentConfiguration` off the raw `GenericConfiguration` map onto
the typed configuration model. The struct is now built from the translated
`shared::ClusterAgent` slice; the `cluster_agent.*` keys are witnessed (present
in the vendored Datadog schema) and already resolved into the model.

Trimming of the Cluster Agent URL, auth token, and Kubernetes service name moves
up into the config layer (the witness translator), since trailing whitespace
would otherwise corrupt the value at the point of use (URL parsing, bearer
token, service-name-derived env lookup). Component construction just copies the
already-normalized fields, and its `from_configuration` signature changes from
`&GenericConfiguration` to `&shared::ClusterAgent`.

The call site in `run.rs` now passes the typed `shared.cluster_agent` slice
alongside the existing `shared.autoscaling_failover` slice.

## Change Type
- [x] Non-functional (chore, refactoring, docs)

## How did you test this PR?

- Component construction tests rewritten to build from `ClusterAgent` model
  slices directly (enabled gating, URL scheme normalization, Kubernetes service
  resolution, IPv6 host wrapping, URL-vs-service precedence).
- Added translator tests covering the relocated normalization: surrounding
  whitespace is trimmed and empty/whitespace-only values collapse to `None`, and
  an absent `kubernetes_service_name` picks up the Datadog schema default.
- `make build-schema-overlay && make fmt` (no file changes), `make check-all`,
  `make test`, `make check-docs` all pass.

### Follow-up Fixed by AI Review

Claude Code operating on behalf of webern.

Follow-up review of `216f6836f89f333773e21fc04032134a2b2729a3`: the reported regression is fixed. The translator now preserves the distinction between the schema default and an explicitly blank `cluster_agent.kubernetes_service_name`: absent input still becomes `Some("datadog-cluster-agent")`, configured values are trimmed, and empty or whitespace-only input becomes `Some("")`. That empty value reaches `resolve_endpoint`, which suppresses Kubernetes service discovery even when the default service environment variables are present. URL and auth-token normalization remain unchanged.

I verified coverage for the schema default, explicit empty input, whitespace-only input, and the component-level suppression behavior with default service environment variables available. The fix is narrowly scoped, and the relevant formatting, Clippy, schema-overlay, and unit-test CI checks reported success during verification.

The original review concern is resolved. This is good to go from the config-cutover review perspective.

## References

- Progresses #1788
- Merges into `m/pr5-cutover`

### CODE REVIEW

File: lib/agent-data-plane-config-system/src/translators/datadog_translator.rs [hunk 1]
```diff
@@ -340,19 +354,19 @@ impl DatadogConfigWitness for DatadogTranslator<'_> {
...
+        self.config.shared.cluster_agent.auth_token = trimmed_non_empty(value);
     }
 
     fn consume_cluster_agent_enabled(&mut self, value: bool) {
         self.config.shared.cluster_agent.enabled = value;
     }
 
     fn consume_cluster_agent_kubernetes_service_name(&mut self, value: String) {
-        self.config.shared.cluster_agent.kubernetes_service_name = non_empty(value);
+        self.config.shared.cluster_agent.kubernetes_service_name = trimmed_non_empty(value);
```
Comment (webern) [inline lib/agent-data-plane-config-system/src/translators/datadog_translator.rs]: When `cluster_agent.kubernetes_service_name` is explicitly empty or whitespace, the old `get_trimmed_string` path produced `Some("")`. `resolve_endpoint` then hit `service_name.is_empty()` and returned `None`, so discovery was suppressed. `trimmed_non_empty` now produces `None`, which this component interprets as “use `datadog-cluster-agent`” through `unwrap_or`. In a Kubernetes deployment with the default service environment variables present, clearing this setting can therefore unexpectedly wire the autoscaling-failover branch to the default service. Preserve `Some("")` for this field (trim without collapsing empty), or represent explicit suppression separately, and add a test covering both empty and whitespace-only values.

**Review by webern — COMMENTED** [review]
Claude Code operating on behalf of webern.\n\nClean-room review found one behavioral regression in explicit `cluster_agent.kubernetes_service_name` handling. The URL and token cutovers preserve the previous trimming and empty-value behavior, and the typed fields remain reachable through the schema/translator. Please preserve the previous empty-service semantics and add a regression test; otherwise this cutover can unexpectedly enable Cluster Agent forwarding to the default Kubernetes service.

**Comment by webern** [comment]
Claude Code operating on behalf of webern.

Follow-up review of `216f6836f89f333773e21fc04032134a2b2729a3`: the reported regression is fixed. The translator now preserves the distinction between the schema default and an explicitly blank `cluster_agent.kubernetes_service_name`: absent input still becomes `Some("datadog-cluster-agent")`, configured values are trimmed, and empty or whitespace-only input becomes `Some("")`. That empty value reaches `resolve_endpoint`, which suppresses Kubernetes service discovery even when the default service environment variables are present. URL and auth-token normalization remain unchanged.

I verified coverage for the schema default, explicit empty input, whitespace-only input, and the component-level suppression behavior with default service environment variables available. The fix is narrowly scoped, and the relevant formatting, Clippy, schema-overlay, and unit-test CI checks reported success during verification.

The original review concern is resolved. This is good to go from the config-cutover review perspective.

File: lib/agent-data-plane-config-system/src/translators/datadog_translator.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (webern) [reply lib/agent-data-plane-config-system/src/translators/datadog_translator.rs]: Fixed with follow up push.

[the remark this answers is not in the document]
Comment (webern) [reply lib/agent-data-plane-config-system/src/translators/datadog_translator.rs]: Fixed with follow up push.

Comment (webern) [reply lib/agent-data-plane-config-system/src/translators/datadog_translator.rs]: Fixed with follow-up push.
